### ISSUE TITLE
Crash if terminal goes from interactive to non-interactive

### ISSUE THREAD

**Issue by jyn514** [oilshell/oil#398]
Steps to reproduce: 
```
(osh) $ exec >> log.txt 2>&1
<tab><tab>
exit
(osh) $ cat log.txt
  File "/usr/local/src/oil/bin/oil.py", line 868, in <module>
    main(sys.argv)
  File "/usr/local/src/oil/bin/oil.py", line 836, in main
    sys.exit(AppBundleMain(argv))
  File "/usr/local/src/oil/bin/oil.py", line 806, in AppBundleMain
    status = ShellMain('osh', argv0, main_argv, login_shell)
  File "/usr/local/src/oil/bin/oil.py", line 611, in ShellMain
    return main_loop.Interactive(opts, ex, c_parser, display, errfmt)
  File "/usr/local/src/oil/core/main_loop.py", line 47, in Interactive
    result = c_parser.ParseInteractiveLine()
  File "/usr/local/src/oil/osh/cmd_parse.py", line 1837, in ParseInteractiveLine
    self._Peek()
  File "/usr/local/src/oil/osh/cmd_parse.py", line 451, in _Peek
    w = self.w_parser.ReadWord(self.next_lex_mode)
  File "/usr/local/src/oil/osh/word_parse.py", line 1231, in ReadWord
    w, need_more = self._ReadWord(lex_mode)
  File "/usr/local/src/oil/osh/word_parse.py", line 1141, in _ReadWord
    self._Peek()
  File "/usr/local/src/oil/osh/word_parse.py", line 120, in _Peek
    self.cur_token = self.lexer.Read(self.next_lex_mode)
  File "/usr/local/src/oil/frontend/lexer.py", line 247, in Read
    t = self._Read(lex_mode)
  File "/usr/local/src/oil/frontend/lexer.py", line 219, in _Read
    line_id, line, line_pos = self.line_reader.GetLine()
  File "/usr/local/src/oil/frontend/reader.py", line 35, in GetLine
    line = self._GetLine()
  File "/usr/local/src/oil/frontend/reader.py", line 78, in _GetLine
    line = raw_input(self.prompt_str) + '\n'  # newline required
IOError: [Errno 25] Inappropriate ioctl for device
```

Found this while investigating a different bug, not sure if this is worth fixing. I think this traceback comes directly from readline, it doesn't look like our code.

**Comment by andychu** [comment]
Yes nice find.  I consider every unhandled stack trace a bug, since we can always at least catch it and abort the shell.

I put a "catchall" around all shell execution a couple months ago.  However it looks like there should be one around parsing, or the entire program.  Parsing gets text from GNU readline, and it's raising exceptions that we're not handling.

**Comment by andychu** [comment]
Released with 0.7.pre1: https://www.oilshell.org/release/0.7.pre1/
